zbench.zig

```diff
@@ -89,12 +92,22 @@ const Definition = struct {
         defer if (self.config.hooks.after_each) |hook| hook();
 
         var t = try std.time.Timer.start();
+        const baseline: u64 = t.read();
+        t.reset();
+
         switch (self.defn) {
             .simple => |func| func(allocator),
             .parameterised => |x| x.func(@ptrCast(x.context), allocator),
         }
+
+        var lap = t.read();
+        if (self.config.baseline_correction) {
+            lap -|= baseline;
```

Comment:
ensure baseline is not greater than lap to prevent negative timing values, or have i overlooked it here?

```suggestion
    if (lap > baseline) {
        lap -= baseline;
    } else {
        lap = 0; // Set lap to 0 if baseline is greater or equal
    }
```

Reply:
since `lap` is a u64, Zig would let this fail if I would cause an overflow (baseline > lap) - that's why I use the [saturating subtraction](https://ziglang.org/documentation/0.12.0/#toc-Operators) `-|` :)

Reply:
Ah okey, forgot to check the syntax here. Then all good 👍